core/kazoo_documents/src/kzd_rate.erl

```diff
@@ -49,18 +49,17 @@ from_map(Map) ->
     ensure_id(set_type(Rate)).
 
 -spec ensure_id(doc()) -> doc().
+-spec ensure_id(doc(), api_ne_binary()) -> doc().
 ensure_id(Rate) ->
-    case kz_doc:id(Rate) of
-        'undefined' -> set_rate_id(Rate);
-        _Id -> Rate
-    end.
+    ensure_id(Rate, kz_json:get_ne_binary_value(<<"_id">>, Rate)).
```

Comment:
Which is fine if _id exists; but if _id is missing but id exists, it won't set _id and you get a 32-char uuid for id.